voicevox_engine/morphing.py

```diff
@@ -43,6 +47,53 @@ def create_morphing_parameter(
     )
 
 
+def is_synthesis_morphing_permitted(
+    engine: SynthesisEngineBase, base_speaker: int, target_speaker: int
+) -> Optional[bool]:
+    """
+    指定されたspeakerがモーフィング可能かどうか返す
+    speakerが見つからない場合はNoneを返す
+    """
+
+    speakers = json.loads(engine.speakers)
+    base_speaker_info, target_speaker_info = None, None
+    for speaker in speakers:
+        style_id_arr = tuple(map(lambda style: style["id"], speaker["styles"]))
+        if base_speaker_info is None and base_speaker in style_id_arr:
+            base_speaker_info = speaker
+        if target_speaker_info is None and target_speaker in style_id_arr:
+            target_speaker_info = speaker
+
+    if base_speaker_info is None or target_speaker_info is None:
+        return None
+
+    base_speaker_morphing_info: SpeakerSupportSynthesisMorphing = base_speaker_info.get(
+        "supported_features", dict()
+    ).get("synthesis_morphing", SpeakerSupportSynthesisMorphing(None))
+
+    target_speaker_morphing_info: SpeakerSupportSynthesisMorphing = (
+        target_speaker_info.get("supported_features", dict()).get(
+            "synthesis_morphing", SpeakerSupportSynthesisMorphing(None)
+        )
+    )
+
+    if (
+        base_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PROHIBIT
+        or target_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PROHIBIT
+    ):
+        return False
+    if (
+        base_speaker_morphing_info == SpeakerSupportSynthesisMorphing.SELF_MORPHING_ONLY
+        or target_speaker_morphing_info
+        == SpeakerSupportSynthesisMorphing.SELF_MORPHING_ONLY
+    ):
+        return base_speaker_info["speaker_uuid"] == target_speaker_info["speaker_uuid"]
+    return (
+        base_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PERMIT
+        and target_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PERMIT
+    )
```

Comment:
意図がとても大事なので、意図をコメントするのも良いかもと思いました！
こんな感じ･･･？
```suggestion
    # 禁止されている場合はFalse
    if (
        base_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PROHIBIT
        or target_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PROHIBIT
    ):
        return False
    # 同一話者のみの場合は同一話者判定
    if (
        base_speaker_morphing_info == SpeakerSupportSynthesisMorphing.SELF_MORPHING_ONLY
        or target_speaker_morphing_info
        == SpeakerSupportSynthesisMorphing.SELF_MORPHING_ONLY
    ):
        return base_speaker_info["speaker_uuid"] == target_speaker_info["speaker_uuid"]
    # 念のため許可されているかチェック
    return (
        base_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PERMIT
        and target_speaker_morphing_info == SpeakerSupportSynthesisMorphing.PERMIT
    )
```